engine-types/src/types/address.rs

```diff
@@ -0,0 +1,144 @@
+use crate::{format, String, TryFrom, H160};
+use borsh::maybestd::io;
+use borsh::{BorshDeserialize, BorshSerialize};
+
+/// Base Eth Address type
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+pub struct Address(H160);
+
+#[allow(non_snake_case, dead_code)]
+pub const fn ADDRESS(addr: H160) -> Address {
+    Address(addr)
+}
```

Comment:
No, we can't. `const fn` is not the same as calling `Address::new`. That method we call in another `const fn`. So we can't do that.

But more importantly, the constant method is about optimization. In many places, we use a 
compilation of code that converts to a constant value. Those. compilation level optimization. 

And we can't use the same `const name` as `type name`. Previously it was just alias to `H160`.

I decided to use more const-specific naming. 

But if we remove `const fn` it will performance regrattion.

Reply:
I'm not saying we should get rid of `const` optimization. I'm saying calling `ADDRESS` as a function is not idiomatic. Either make `const fn new` or simply make the field public (tuple constructors are always `const`).